azurerm/internal/services/keyvault/resource_arm_key_vault.go

```diff
@@ -374,6 +589,20 @@ func resourceArmKeyVaultDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error retrieving Key Vault %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
+	if read.Properties == nil {
+		return fmt.Errorf("Error retrieving Key Vault %q (Resource Group %q): `properties` was nil", name, resourceGroup)
+	}
+
+	// Check to see if purge protection is enabled or not...
+	if read.Properties == nil {
```

Comment:
This entire `if` condition is not necessary since  the chunk of code directly above this already checks for the `nil` condition.... the `if` condition should be removed, but the code between the `if` should remain...